wiki/Publishing-guidelines/Writing-functional-specs.md

```diff
@@ -0,0 +1,383 @@
+
+<br />
+</details><br /><br />
+
+#### Used by
+
+By properly linking dependencies we will also run an automation script to collect all the dependencies and create a used by list in addition. Used by components once populated communicates to the reader where this component is used.
+
+###### How to add a used by
+
+> ![image](https://user-images.githubusercontent.com/3793636/117873919-f6faba80-b265-11eb-81a5-039bdcd822e8.png)  Keep dependencies up to date, and leverage used by template section in the component wiki page.
+
+#### Backlinks
+Every time a wiki document references a component other than itself it creates a back link. Those back links are listed out on that component's page as a snippet and categorized by pages. When clicked it will go to the section of the page where it was talked about, and highlight its contents. A backlink doesn't have to be used as a depency to work, and this list is automated. As long as you include it on the page it'll render its content if available.
+
+#### Resources
+The resource section is a numbered list of additional resources not included in the dependency list. For example design specs, box folders, prototypes, and maybe additional documentation are good things you could include here. Follow the numbering standard if possible to star consistent. `r` for resources, and `#` in the list. If for whatever reason we need to remove an item it's best to keep the numbers the name, and skip since there is possible cross page linking and to avoid multiple updates when a resource becomes outdated.
+
+<details>
+  <summary><strong>Example</strong></summary><br />
+
+```html
+- r1: [resource example](www.example.com)
+- r2: [resource example](www.example.com)
+- r3: [resource example](www.example.com)
+- r6: [resource example](www.example.com)
+- r7: [resource example](www.example.com)
+```
+
+<br />
+</details><br /><br />
+
+### Sections
+
+The sections on the page are just some ways I've used in the past to break content down, and provide the functional specs required for development. Ultimately the following levels use at your discretion, and discuss in review, but the following is how I tend to like to use them.
+
+<br /><br />
+
+#### Global
+The global level is often where I'll put a visual of the component, and a high level break down of its elements and components. Every once in while I'll add a global option table or a logic conditional table to demonstrate how whole entire component's functionality can be modified. I typically won't include a markdown title to help indicate that it's apart of the highest level of the component.
+
+#### Level 1
+Level 1 tends to be the layer of elements, and usually any dependencies if there are any. Here I can identify what high level elements are `optional`, and what `type` (e.g. `string`, `component`) it might be.
```

Comment:
```suggestion
Level 1 usually covers the layer of elements, along with any dependencies. This is where you identify what high level elements are `optional`, and what `type` (e.g. `string`, `component`) it might be.
```

Reply:
I think I meant to add the word "highest" here? And maybe "outline" makes more sense than covers? 

> Level 1 typically outlines the highest layer of elements within a component, along with any dependencies. This is where you identify what high level elements are `optional`, and what `type` (e.g. `string`, `component`) it might be. Underneath here we can continue to break it down further, or add functional details as needed.

Reply:
Yes this is great, James. For this sentence: 

> Underneath here we can continue to break it down further, or add functional details as needed.

Do you mean in this section? If so, does this work?

> Use this section to break the content down further, or add functional details, as needed.

Reply:
I might would suggest we use the word component instead of content because it can be a little more than that, but yeah that works for me.